src/common/main/kotlin/tech/thatgravyboat/skyblockapi/api/remote/api/SimpleItemAPI.kt

```diff
@@ -72,45 +89,64 @@ object SimpleItemAPI {
     }
 
     fun getRuneByIdOrNull(id: SkyBlockId): ItemStack? = cache.getOrPut(id.trySafe(::rune)) {
-        val runeId = id.toApiIdOrNull() ?: return@getOrPut null
+        val runeId = id.cleanId.uppercase()
+
+        if (runeId == UNKNOWN) {
+            return@getOrPut null
+        }
 
         if (runeId.contains(":")) {
             val (runeId, literalLevel) = runeId.split(":")
             val level = literalLevel.toIntValue()
             RepoRunesAPI.getRuneAsItemOrNull(runeId, level)?.let { return@getOrPut it }
         }
 
-        RepoRunesAPI.getRuneAsItemOrNull(runeId.substringBefore(":"))
+        return@getOrPut RepoRunesAPI.getRuneAsItemOrNull(runeId.substringBefore(":"), null)
     }
 
     fun getRuneById(id: SkyBlockId) = getRuneByIdOrNull(id) ?: ItemBuilder(Items.BARRIER) {
         name("Unknown rune: $id")
     }
 
     fun getEnchantmentByIdOrNull(id: SkyBlockId): ItemStack? = cache.getOrPut(id.trySafe(::enchantment)) {
-        val enchantmentId = id.toApiIdOrNull() ?: return@getOrPut null
+        val enchantmentId = id.cleanId.uppercase()
+
+        if (enchantmentId == UNKNOWN) {
+            return@getOrPut null
+        }
 
         if (enchantmentId.contains(":")) {
             val (enchantmentId, literalLevel) = enchantmentId.split(":")
             val level = literalLevel.toIntValue()
             RepoEnchantmentAPI.getEnchantmentAsItemOrNull(enchantmentId, level)?.let { return@getOrPut it }
         }
 
-        RepoEnchantmentAPI.getEnchantmentAsItemOrNull(enchantmentId.substringBefore(":"))
+        return@getOrPut RepoEnchantmentAPI.getEnchantmentAsItemOrNull(enchantmentId.substringBefore(":"), null)
     }
 
     fun getEnchantmentById(id: SkyBlockId): ItemStack = getEnchantmentByIdOrNull(id) ?: ItemBuilder(Items.BARRIER) {
         name("Unknown enchantment: $id")
     }
 
     fun getAttributeByIdOrNull(id: SkyBlockId): ItemStack? = cache.getOrPut(id.trySafe(::attribute)) {
-        id.toApiIdOrNull()?.let(RepoAttributeAPI::getAttributeByIdOrNull)
+        val attributeId = id.cleanId.uppercase()
+
+        if (attributeId == UNKNOWN) {
+            return@getOrPut null
+        }
+
+        RepoAttributeAPI.getAttributeByIdOrNull(attributeId)?.let { return@getOrPut it }
+
+        return@getOrPut null
```

Comment:
cant this just be simplified to
```suggestion
        return RepoAttributeAPI.getAttributeByIdOrNull(attributeId)
```